salt/profiles/init.sls

```diff
@@ -64,7 +71,8 @@ profiles-uwsgi-service:
         - require:
             - file: profiles-uwsgi-service
         - watch:
-            - profiles-install
+            - file: /srv/profiles*
```

Comment:
This means that it will only restart when something changes.

Reply:
catches the `file.managed` states but not `profiles-install` which is a `cmd`?

Reply:
It's watching the directory itself, so any file change triggers it. Tested by changing one of the config files.

Reply:
As far as I know the watch is expanded to all state ids/names at compile time, rather than to the actual content of the server. Demonstration:
```
$ ps faxww | grep uwsgi
  808 ?        Ss     0:01 /srv/profiles/venv/bin/uwsgi --die-on-term --ini /srv/profiles/uwsgi.ini
 2518 ?        S      0:00  \_ /srv/profiles/venv/bin/uwsgi --die-on-term --ini /srv/profiles/uwsgi.ini
 4195 pts/3    S+     0:00              \_ grep --exclude=*.pyc --color=auto uwsgi
$ rm -rf venv/
(run highstate)
$ ps faxww | grep uwsgi
  808 ?        Ss     0:01 /srv/profiles/venv/bin/uwsgi --die-on-term --ini /srv/profiles/uwsgi.ini
 2518 ?        S      0:00  \_ /srv/profiles/venv/bin/uwsgi --die-on-term --ini /srv/profiles/uwsgi.ini
 5274 pts/3    S+     0:00              \_ grep --exclude=*.pyc --color=auto uwsgi
$ ls venv/
bin  include  lib  pip-selfcheck.json
```
because `venv` is not covered (as wouldn't anything created by `install.sh` I think)

Reply:
Interesting. But is it not ok, as `venv` etc would only change when something in the repo has changed?

Reply:
Yeah, not a bad idea to always restart. But we shouldn't use `watch` to do so, let's make it explicit.